diff --git a/core/dpdk.c b/core/dpdk.c
index add6951..85be3e8 100644
--- a/core/dpdk.c
+++ b/core/dpdk.c
@@ -111,6 +111,8 @@ static void init_eal(char *prog_name, int mb_per_socket, int multi_instance)
 #if 1
 	rte_argv[rte_argc++] = "--socket-mem";
 	rte_argv[rte_argc++] = opt_socket_mem;
+	rte_argv[rte_argc++] = "-w";
+	rte_argv[rte_argc++] = "00:07.0";
 #else
 	rte_argv[rte_argc++] = "--no-huge";
 #endif
diff --git a/core/namespace.c b/core/namespace.c
index 90820ac..0633621 100644
--- a/core/namespace.c
+++ b/core/namespace.c
@@ -194,7 +194,7 @@ int ns_is_valid_name(const char *name)
 
 	for (int i = 1; i < name_len; i++) {
 		c = name[i];
-		if ((c != '_') && !isalnum(c))
+		if ((c != '_') && !isalnum(c) && c != '-')
 			return 0;
 	}
 
